src/sonic-yang-models/yang-models/sonic-mclag.yang

```diff
@@ -0,0 +1,240 @@
+                        }
+                    }
+                }
+                leaf keepalive_interval {
+                    type uint16 {
+                        range "1..60" {
+                            error-message "MCLAG Domain keepalive interval out of range";
+                            error-app-tag keepalive_interval-invalid;
+                        }
+                    }
+                    default 1;
+                }
+                leaf session_timeout {
+                    type uint16 {
+                        range "1..3600" {
+                            error-message "MCLAG Domain session timeout out of range";
+                            error-app-tag session_timeout-invalid;
+                        }
+                    }
+                    default 30;
+                }
+                must "(keepalive_interval * 3) <= session_timeout" {
+                        error-message "(keepalive interval * 3) <= session_timeout value";
+                        error-app-tag keepalive_interval-invalid;
+                }
+            }
+        }
+
+        container MCLAG_INTERFACE { 
+
+            list MCLAG_INTERFACE_LIST { 
+                key "domain_id if_name";
+
+                leaf domain_id {
+                    type leafref {
+                        path "../../../MCLAG_DOMAIN/MCLAG_DOMAIN_LIST/domain_id";
+                    }
+                }
+
+                leaf if_name {
+                    type leafref {
+                        path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                    }
+                }
+                leaf if_type {
+                    type string;
+                }
+            }
+        }
+
+        container MCLAG_UNIQUE_IP {
+
+            list MCLAG_UNIQUE_IP_LIST {
+                key "if_name";
+
+                must "count(../../MCLAG_DOMAIN/MCLAG_DOMAIN_LIST/domain_id) != 0" {
+                    error-message "mclag not configured";
+                    error-app-tag mclag-invalid;
+                }
+
+                must
+                    "count(/svlan:sonic-vlan/svlan:VLAN_INTERFACE/svlan:VLAN_INTERFACE_IPPREFIX_LIST[svlan:name=current()/if_name]) = 0" {
+                    error-message "remove configured ip/v6 address and reconfigure after separate ip configuration";
+                    error-app-tag vlan-intf-ip-invalid;
+                }
+
+                leaf if_name {
+                    type leafref {
+                        path "/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST/svlan:name";
+                    }
+                }
+                leaf unique_ip {
+                    type enumeration {
+                        enum enable;
+                    }
+                }
+            }
+        }
+
+        //stateDB MCLAG Table
+        container MCLAG_TABLE { 
+            sonic-ext:db-name "STATE_DB"; 
+            sonic-ext:key-delim "|"; 
+
```

Comment:
Not sure if we need to add any model for STATE_DB now, please remove it.

Reply:
STATE_DB entry are required for show commands

Reply:
We have not added any STATE_DB tables in the YANG model so far, this is something we need to discuss and decide in the community meeting, if required, we need to plan for adding other STATE_DB tables as well.

Reply:
thanks for the comments. currently sonic-extension changes for db-name, key-delimit etc is not yet into community. so i will remove state db entries which requires this and add it later when those are available